### PR TITLE
Create a separate file for a portable GeneralFormat

### PR DESCRIPTION
**Pull request by stevedlawrence**
The DFDLGeneralFormat.dfdl.xsd file defines the emptyElementParsePolicy extension property in the GeneralFormat format. Even if a schema uses GeneralFormatPortable, which does not deine this exension property, IBM DFDL still sees it and creates an error. This means that it is impossible to use this file for portability tests.

To resolve this, this creates a new DFDLGeneralFormatPortable.dfdl.xsd file that does not contain any properties that IBM DFDL does not support. Additionally, the define format in this file uses the same GeneralFormat name as DFDLGeneralFormat.dfdl.xsd. This allows schema authors to alway reference the GeneralFormat format, and only change the xs:include to make a schema portable or not.

Compatibility/Deprecation:

The GeneralFormatPortable define format has been removed. Instead, portable schemas should import a new portable file and reference the "GeneralFormat" format. For example, a schema that previously looked like this:

    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd" />

    <dfdl:format ref="GeneralFormatPortable" ... />

Should now look like this:

    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormatPortable.dfdl.xsd" />

    <dfdl:format ref="GeneralFormat" ... />

DAFFODIL-2820

### CODE REVIEW

**Comment by stevedlawrence** [comment]
Note that a number of tests will fail until https://github.com/apache/daffodil/pull/1032 is merged, but I think this is straightforward enough that it an still be reviewed.

File: daffodil-core/src/test/scala/org/apache/daffodil/core/api/TestDsomCompiler3.scala
Comment (tuxji) [inline daffodil-core/src/test/scala/org/apache/daffodil/core/api/TestDsomCompiler3.scala:76]: For other reviewers, this line has been simplified to get the first schemaDoc in a more straightforward way.

File: daffodil-lib/src/main/resources/org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd
Comment (tuxji) [inline daffodil-lib/src/main/resources/org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd:34]: I see GeneralFormatOriginal has been moved to its own file called DFDLGeneralFormatBase.dfdl.xsd and that file included above here.

File: daffodil-lib/src/main/resources/org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd
Comment (tuxji) [inline daffodil-lib/src/main/resources/org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd:110]: Likewise, I see GeneralFormatPortable has been moved to its own file called DFDLGeneralFormatPortable.dfdl.xsd.

**Review by tuxji — APPROVED** [review]
+1, looks good

File: daffodil-core/src/test/scala/org/apache/daffodil/core/api/TestDsomCompiler3.scala
Comment (stevedlawrence) [reply daffodil-core/src/test/scala/org/apache/daffodil/core/api/TestDsomCompiler3.scala:76]: Yep, because there is now an extra include (GeneralFormatBase.xsd) there are now three docs in the `schema.schemaDocuments` list in a lot of tet. The first is still the root schema that we actually care about, but the pattern match fails. I figure switching to using `.head` will allow us to change the includes again if needed without breaking these tests, for example if IBM DFDL and Daffodil fix compatibility issues and the portable stuff goes away.

**Review by mbeckerle — APPROVED** [review]
+1

**Comment by stevedlawrence** [comment]
Yep, I'll wait until the schematron change is merged, rebase this PR to double check all tests pass, and then merge.

**Pull request merged**
